File: pset6/caesar/caesar.py
from cs50 import get_string, sys
#print(len(sys.argv[1]))
#print(type(sys.argv[1]))
if len(sys.argv) <= 1 or len(sys.argv) > 2:
    print("Usage: python caesar.py k")
    sys.exit(1)
key = int(sys.argv[1])
text = get_string("plaintext: ")
print("ciphertext: ", end="")
for i in range(len(text)):
    if text[i].isalpha():
        #print(text[i])
        if text[i].isupper():
            #print(text[i])
            letter = ord(text[i]);
            result = (((((letter -65) + key) % 26) + 65))
            print(chr(result), end="")
        elif text[i].islower():
            #print(text[i])
            letter = ord(text[i])
            result = ((((letter -97) + key) % 26) + 97)
            print(chr(result), end="")
    else:
        print(text[i], end="")
print("")
#print(int(sys.argv[1]))

File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    //checking user pass only one argument
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    // checking only pass numbers
    string clave = argv[1];
    for (int i = 0; i < strlen(clave); i++)
    {
        if (isdigit(clave[i]) == 0)
        {
            return 1;
        }
    }
    //convert string to int
    int key = atoi(clave);

    string text = get_string("plaintext:  ");

    printf("ciphertext: ");

    //cipher the text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            //checking uppercase
            if (isupper(text[i]))
            {
                int letter = text[i];
                int result = ((((letter - 65) + key) % 26) + 65);
                printf("%c", result);
            }
            //checking lowercase
            else if (islower(text[i]))
            {
                int letter = text[i];
                int result = ((((letter - 97) + key) % 26) + 97);
                printf("%c", result);
            }
        }
        //else print that character
        else
        {
            printf("%c", text[i]);
        }
    }
    //return 0 if successful
    printf("\n");
    return 0;
}

